### DISCUSSION TITLE
Design & Implementation Stand-ups

### DISCUSSION THREAD

**Discussion by eshiofune** [eshiofune/PromptXchange#6] [General]
### Guidelines

**What you worked on this past week**

**What will you work on next week?**

**Any Blockers?**

**Comment by eshiofune** [comment]
What you worked on this past week

- Setup a Django project
- Setup docker for the project
- Added setup instructions to the readme
- Added a config file

What will you work on next week?

- Create models for managing tasks and prompts
- Create/update the user model
- Create an endpoint (possibly setup Django Rest Framework) for submitting a new task/prompt pair to the db
- Create an endpoint to request a prompt for a task (possibly using RAG with the "documents" being the task/prompt pairs in the db)
- Start designing /  working on an LLM interface that would allow a user teach the bot how to perform a task, and at the end of the session, the training would be summarised and added to the db as a task/prompt pair.

Any Blockers?

No

**Comment by dumb0ctopus** [comment]
What I did last week:

- Explored various methods for implementing a search function on the website.
- Acquired the foundational code for the search functionality.

What I'll do next week:

- Research and implement best practices for collaborating on software projects.
- Establish contribution guidelines to streamline collaboration efforts.
- Develop the frontend folder and file structure for the project.
- Define all requirements and features for the search functionality to ensure comprehensive implementation.

Any Blockers?

No

**Comment by catladydesignr** [comment]
What I did last week?

- Started Branding (color pallete and font)
- Designed the read-me file

What I will do next week:

- Make Logo 
- User research and flows




Any Blockers?

No

**Comment by eshiofune** [comment]
@catladydesignr 

What i did last week?

- Worked on user flows
- created user personas
- Design research
- Features list (still in progress)

What i will do next week?

- Moodboards
- Wireframes
- Contributor level/tier configuration
- Search bar text limit for search suggestion

Blockers?
Yes, creative block i shall be moving this indefinitely

**Comment by eshiofune** [comment]
What you worked on this past week
- Create models for managing tasks and prompts

What will you work on next week?

- Continue with creating models for managing tasks and prompts
- Create/update the user model
- Create an endpoint (possibly setup Django Rest Framework) for submitting a new task/prompt pair to the db
- Create an endpoint to request a prompt for a task (possibly using RAG with the "documents" being the task/prompt pairs in the db)
- Start designing / working on an LLM interface that would allow a user teach the bot how to perform a task, and at the end of the session, the training would be summarised and added to the db as a task/prompt pair.

Any Blockers?

No

**Comment by dumb0ctopus** [comment]
What you worked on this past week

- Developed the frontend folder and file structure for the project.
- Defined all requirements and features for the search functionality to ensure comprehensive implementation.
- Researched best practices for collaborating on software projects.

What will you work on next week?

- Establish contribution guidelines to streamline collaboration efforts.
- Begin work on the search bar.

Any Blockers?

No

**Comment by eshiofune** [comment]
What you worked on this past week

- [x]  Created models for managing tasks and prompts, for user accounts and api keys which will be used to connect to the Prompt xChange API
- [x]  Create/update the user model
- [x]  setup Django Rest Framework with permissions, serialisers and viewsets

What will you work on next week?

- Create an endpoint for submitting a new task/prompt pair to the db
- Create an endpoint to request a prompt for a task (possibly using RAG with the "documents" being the task/prompt pairs in the db)
- Start designing / working on an LLM interface that would allow a user teach the bot how to perform a task, and at the end of the session, the training would be summarised and added to the db as a task/prompt pair.

Any Blockers?

No

**Comment by dumb0ctopus** [comment]
What you worked on this past week

- Built a functional search bar that can give a maximum of 5 most relevant results, as the user is typing.

What will you work on next week?

- Finetune the search bar.
- Build a basic chatbot interface.

Any Blockers?

No.
